src/Nethermind/Nethermind.AuRa/IAuRaAdditionalBlockProcessorFactory.cs

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2018 Demerzel Solutions Limited
+ * This file is part of the Nethermind library.
+ *
+ * The Nethermind library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Nethermind library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+using Nethermind.AuRa.Validators;
+using Nethermind.Core.Specs.ChainSpecStyle;
+
+namespace Nethermind.AuRa
+{
+    public interface IAuRaAdditionalBlockProcessorFactory
+    {
+        IAuRaValidatorProcessor CreateValidatorProcessor(AuRaParameters.Validator validator, long? startBlock = null);
```

Comment:
reading forward but I am interested in why the startBlock is needed here?

Reply:
For contract validators (but abstracted here to potentially any validator):
* To know when to initialize yourself - contract validators need to call FinalizeChange after they are initialized.
* Not every contract validator will be active at block 1, when we have multi validator (switching contract validators).